src/shared/styles/variables/palette.ts

```diff
@@ -1,86 +1,274 @@
-export const palette = {
-  primary: '#00639a',
-  primary50: '#207cbb',
-  primary80: '#96ccff',
-  primary95: '#e8f2ff',
-  primary_container: '#cee5ff',
-  secondary: '#51606f',
-  secondary60: '#8392a3',
-  secondary_container: '#d5e4f7',
-  surface1: '#eff4fa',
-  surface2: '#e8f0f7',
-  surface3: '#e0ebf4',
-  surface5: '#d9e7f1',
-
-  on_surface: '#1a1c1e',
-  on_surface_variant: '#42474e',
-  outline: '#72777f',
-  outline_variant: '#c4c9d0',
+import { flattenPaletteObject, hexToRgba } from '../theme.utils';
+
+type BasePalette = Record<string, Record<string | number, string>>;
+export interface Palette {
+  [key: string]: string;
+}
+
+/**
+ * This references all base "primitive" colors.
+ * These colors will be used to give values to
+ * semantic colors, like "on_container".
+ */
+const basePalette: BasePalette = {
+  primary: {
+    // Main primary color falls under primary[3]
+    // For the rest of the colors, the default value is [40]
+    default: '#0B0907',
+    0: '#000',
+    10: '#1D1B19',
+    20: '#32302D',
+    30: '#484744',
+    40: '#5F5E5B',
+    50: '#787773',
+    60: '#91918E',
+    70: '#ACABA9',
+    80: '#C6C6C5',
+    90: '#E3E2E1',
+    95: '#F1F0EF',
+    99: '#FDFCFC',
+    100: '#FFF',
+  },
+  secondary: {
+    0: '#000',
+    10: '#041847',
+    20: '#182D65',
+    30: '#173F9F',
+    40: '#004CED',
+    50: '#216DFF',
+    60: '#3C90FF',
+    70: '#55B1FF',
+    80: '#8BCEFF',
+    90: '#CAE6FF',
+    95: '#E5F3FF',
+    99: '#FBFCFF',
+    100: '#FFF',
+  },
+  tertiary: {
+    0: '#000',
+    10: '#320F1D',
+    20: '#4B2432',
+    30: '#6B364A',
+    40: '#944262',
+    50: '#AF5C82',
+    60: '#CA779E',
+    70: '#D69AB8',
+    80: '#E5BCD1',
+    90: '#F0DAE5',
+    95: '#F9EEF4',
+    99: '#FFFBFF',
+    100: '#FFF',
+  },
+  error: {
+    0: '#000',
+    10: '#410002',
+    20: '#690004',
+    30: '#930009',
+    40: '#BA1A1A',
+    50: '#DE3730',
+    60: '#FF5449',
+    70: '#FE897D',
+    80: '#FFB4AB',
+    90: '#FFDAD6',
+    95: '#FFEDEA',
+    99: '#FFFBFF',
+    100: '#FFF',
+  },
+  neutral: {
+    0: '#000',
+    10: '#1D1B19',
+    20: '#32302D',
+    30: '#484744',
+    40: '#5F5E5B',
+    50: '#787773',
+    60: '#91918E',
+    70: '#ACABA9',
+    80: '#C6C6C5',
+    90: '#E3E2E1',
+    95: '#F1F0EF',
+    99: '#FDFCFC',
+    100: '#FFF',
+  },
+  neutral_variant: {
+    0: '#000',
+    10: '#1D1B19',
+    20: '#32302D',
+    30: '#484744',
+    40: '#5F5E5B',
+    50: '#787773',
+    60: '#91918E',
+    70: '#ACABA9',
+    80: '#C6C6C5',
+    90: '#E3E2E1',
+    95: '#F1F0EF',
+    99: '#FDFCFC',
+    100: '#FFF',
+  },
+  surface: {
+    1: '#F5F4F4',
```

Comment:
I think this one is missing (`palette.surface` is `undefined`):
```suggestion
    default: '#FDFCFC',
    1: '#F5F4F4',
```
![CleanShot 2025-06-16 at 20 07 28@2x](https://github.com/user-attachments/assets/769bfdc3-49a6-4e72-8eb6-c5c097ef7008)

Reply:
Since this one is tied directly to the neutral palette, I didn't define it here, but rather on L193 (semanticPalette object)

Reply:
Ref here (from Figma variables):
![58902](https://github.com/user-attachments/assets/d70998b0-faae-4659-ad64-393d6168a6ea)